lms/djangoapps/courseware/date_summary.py

```diff
@@ -0,0 +1,250 @@
+"""
+This module provides date summary blocks for the Course Info
+page. Each block gives information about a particular
+course-run-specific date which will be displayed to the user.
+"""
+import pytz
+
+from datetime import datetime
+from django.core.urlresolvers import reverse
+from django.utils.translation import ugettext as _
+from edxmako.shortcuts import render_to_string
+from lazy import lazy
+
+from course_modes.models import CourseMode
+from verify_student.models import VerificationDeadline, SoftwareSecurePhotoVerification
+from student.models import CourseEnrollment
+
+
+# pylint: disable=missing-docstring
+
+
+class DateSummary(object):
+    """Base class for all date summary blocks."""
+
+    # The state of this summary. Used visually to indicate the type of
+    # information this summary block contains, and its urgency.
+    state = ''
```

Comment:
Good point; I'd originally thought this might be useful for something else. I'll update it.